### PR TITLE
Incin chems rebalance

### PR DESCRIPTION
**Pull request by jroinc**
## About The Pull Request

Chemical fire now functions properly. WP smoke can now be made chemically. Might look at explosives... but I think that's well-covered by ungas already.

Pre-assembled chem grenades now have a 2-tile radius, same as magic grenades.
Large beaker chem grenades can be made with up to a three-tile radius, one larger than magic fire grenades and one smaller than magic WP grenades.

WP smoke can be made via chemical reaction. A large beaker grenade is about as large as a WP grenade, but has no fire.

Nerfs fuel-air smoke. Radius was too large. It's still big but not "two screens" big.

Adds gunpowder. Chem on par with normal grenades, can be made slightly larger than a M15, but leaves a brief cloud of smoke upon detonation.

Adds ANFO. Chem with heavy explosion, limited to two tiles max, smaller than a normal grenade.

Napalm, WP, gunpowder, and ANFO can be mixed and matched as desired, and will function independently in the same grenade, so you can have explosive fire, mini-WP, a shaped-charge with more damage at the center, or just throw a cloud of WP onto all your normal grenades. Experiment!

Fuel-air mix does *not* mix with other chems, light it with another action. I could fix this, but I'd need to make a snowflake chem and it wouldn't fix actual fuel-air mix so I left it as is. Still very strong VS open terrain.

## Why It's Good For The Game

Chemical fire's been broken for way too long, and it's dumb that the incin grenade spawns one tile of fire. More chemical content and things for people to do better.

## Changelog
:cl:
add: adds chemical WP gas, gunpowder, and ANFO
balance: lowers maximum radius of chemsmoke to around WP-radius. Also raises minimum amount of reactable razorburn to 5/5, from 1/1. 
fix: fixes napalm and napalm grenades
/:cl:

### CODE REVIEW

**Comment by jroinc** [comment]
Testing complete, functions as intended, no further changes unless requested.

**Comment by SplinterGP** [comment]
image

File: code/modules/reagents/reactions/other.dm [hunk 1]
```diff
@@ -275,3 +225,92 @@
...
+					M.Stun(10 SECONDS)
+
+
+/datum/chemical_reaction/napalm
+	name = "Napalm"
+	required_reagents = list(/datum/reagent/aluminum = 1, /datum/reagent/toxin/phoron = 2, /datum/reagent/toxin/acid = 1 )
+
+/datum/chemical_reaction/napalm/on_reaction(datum/reagents/holder, created_volume, radius)
+	radius = round(sqrt(created_volume * 0.15)) //allows a nice, healthy 3-tile fire if using 2 120u beakers fully filled up.
+	if(radius < 0) radius = 0
```
Comment (BraveMole) [inline code/modules/reagents/reactions/other.dm:271]:
```suggestion
	if(radius < 0)
		radius = 0
```

Comment (BraveMole) [reply code/modules/reagents/reactions/other.dm:271]: but what is flame_radius(0) doing?

File: code/modules/reagents/reactions/other.dm [hunk 2]
```diff
@@ -275,3 +225,92 @@
...
+	explosion(get_turf(holder.my_atom), light_impact_range = radius, small_animation = TRUE)
+
+
+/datum/chemical_reaction/explosive/anfo
+	name = "ANFO"
+	required_reagents = list(/datum/reagent/ammonia = 1, /datum/reagent/fuel = 3)
+
+/datum/chemical_reaction/explosive/anfo/on_reaction(datum/reagents/holder, created_volume)
+	var/radius = round(sqrt(created_volume* 0.25), 1) // should be a max of 2 tiles
+	if(radius > 2) radius = 2 //enforced by a hardcap. Sorry!
```
Comment (BraveMole) [inline code/modules/reagents/reactions/other.dm:314]:
```suggestion
	if(radius > 2)
		radius = 2 //enforced by a hardcap. Sorry!
```

**Review by BraveMole — CHANGES_REQUESTED** [review]

File: code/modules/reagents/reactions/other.dm (hunk 1 above)
Comment (jroinc) [reply code/modules/reagents/reactions/other.dm:271]: I don't know why I left that there. It's not like it can be *less* than 0.

Comment (BraveMole) [reply code/modules/reagents/reactions/other.dm]: Ok, so what is flame radius (0) actually doing?

File: code/modules/reagents/reactions/other.dm [hunk 3]
```diff
@@ -275,3 +225,97 @@
...
+/datum/chemical_reaction/emp_pulse
+	name = "EMP Pulse"
+	required_reagents = list(/datum/reagent/uranium = 1, /datum/reagent/iron = 1) // Yes, laugh, it's the best recipe I could think of that makes a little bit of sense
+
+/datum/chemical_reaction/emp_pulse/on_reaction(datum/reagents/holder, created_volume)
+	var/location = get_turf(holder.my_atom)
+	// 100 created volume = 4 heavy range & 7 light range. A few tiles smaller than traitor EMP grandes.
+	// 200 created volume = 8 heavy range & 14 light range. 4 tiles larger than traitor EMP grenades.
+	empulse(location, round(created_volume / 24), round(created_volume / 14), 1)
+	holder.clear_reagents()
```
[the remark this answers is not in the document]
Comment (BraveMole) [reply code/modules/reagents/reactions/other.dm:242]: It's not if tivi feels like removing it, it's either:
It's useful, it stays there.
It's not doing anything, you remove

And it's not up to us to tell you which is which

[the remark this answers is not in the document]
Comment (jroinc) [reply code/modules/reagents/reactions/other.dm:242]: It's useful because it's a chemical reaction that causes an EMP, and is currently the only way to create an EMP in game.

It's not useful because we don't use EMP's for anything, and the reaction is difficult to obtain. 
Both are true. I was leaving it in because I didn't alter it (besides moving it to group it better), and it's something I'm not smart enough to redo easily if we want it for anything.

Basically, I have no strong feelings on the matter.

[the remark this answers is not in the document]
Comment (BraveMole) [reply code/modules/reagents/reactions/other.dm:242]: "t's not useful because we don't use EMP's for anything"

Ok, remove

[the remark this answers is not in the document]
Comment (jroinc) [reply code/modules/reagents/reactions/other.dm]: k

**Review by BraveMole — APPROVED** [review]

**Review by BraveMole — APPROVED** [review]
